pyodide-build/pyodide_build/common.py

```diff
@@ -335,11 +360,8 @@ def get_pyodide_root() -> Path:
 
 @functools.cache
 def get_unisolated_packages() -> list[str]:
-    import json
-
-    if "UNISOLATED_PACKAGES" in os.environ:
```

Comment:
This will make us recompute `unisolated_packages` by scanning every `meta.yaml` file once for every compilation command? There's a reason for this caching logic... `load_all_recipes` is expensive and `functools.cache` doesn't work across subprocesses.

I think we should add a cross-process cache function for this and `get_build_environment_vars` that caches into an environment variable.

Reply:
Good point. I didn't realize that this was for subprocesses. Will revert this change.

Reply:
> every compilation command?

Wait, but is this true? I think we are calling it [once per each package when installing requirements](https://github.com/pyodide/pyodide/blob/4a0ca4ed48ff109679df4e246e8ed79901099f37/pyodide-build/pyodide_build/pypabuild.py#L104).

Reply:
Ah, yes. `pywasmcross.py` looks very clean actually: only thing it uses from the rest of the package is `_f2c_fixes.py`.